examples/pebbleui/js/modules/PebbleUI.js

```diff
@@ -0,0 +1,561 @@
+    circleElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_fill_color(ctx, this._backgroundColor);
+      rocky.graphics_context_set_stroke_color(ctx, rocky.GColorClear);
+      rocky.graphics_fill_radial(ctx, this.bounds, rocky.GOvalScaleModeFitCircle,
+                                Math.min(this.bounds[2], this.bounds[3]) / 2,
+                                0, rocky.DEG_TO_TRIGANGLE(360));
+    };
+
+    return circleElement;
+  };
+
+  // A Text is an Element
+  this.Text = function(options) {
+    options = options || {};
+
+    var textElement = new parent.Element(options);
+
+    textElement._text = options.text || '';
+    textElement._font = rocky.fonts_get_system_font(options.font) ||
+                        rocky.fonts_get_system_font(parent._defaultFont);
+    textElement._color = options.color || rocky.GColorBlack;
+    textElement._alignment = options.alignment || rocky.GTextAlignmentLeft;
+
+    textElement.set = function(data) {
+      if (typeof data === 'string') {
+        this._text = data;
+      } else {
+        for (var key in data) {
+          this._text[key] = data[key];
+        }
+      }
+
+      parent.mark_dirty();
+    };
+
+    textElement._render = function(ctx, bounds) {
+      rocky.graphics_context_set_text_color(ctx, this._color);
+      rocky.graphics_draw_text(ctx, this._text, this._font, this.bounds,
+                                    rocky.GTextOverflowModeWordWrap,
+                                    this._alignment, null);
+    };
+
+    return textElement;
+  };
+
+  // An Image is an Element
+  this.Image = function(options) {
+    options = options || {};
+
+    var imageElement = new parent.Element(options);
+
+    imageElement._bitmap = rocky.gbitmap_create(options.url || '');
+    imageElement._bitmap.onload = function() { parent.mark_dirty(); };
+
+    imageElement._render = function(ctx, bounds) {
+      rocky.graphics_draw_bitmap_in_rect(ctx, this._bitmap, this.bounds);
+    };
+
+    return imageElement;
+  };
+
+  // A Window is an Element
+  this.Window = function(options) {
+    options = options || {};
+
+    var windowElement = new parent.Element(options);
+
+    windowElement._currentZ = 0;
+    windowElement._background = new parent.Rect(options);
+    windowElement._elements = [];
+
+    windowElement.show = function() {
+      var windowIndex = parent._windows.indexOf(this);
+        // Remove the window from the stack if it exists
+      if (windowIndex >= 0) {
+        parent._windows.splice(windowIndex, 1);
+      }
+
+      parent._windows.push(this);
+      parent.mark_dirty();
+      return this;
+    };
+
+    windowElement.hide = function() {
+      var windowIndex = parent._windows.indexOf(this);
+      // Remove the window from the stack if it exists
+      if (windowIndex >= 0) {
+        parent._windows.splice(windowIndex, 1);
+      }
+
+      parent.mark_dirty();
+      return this;
+    };
+
+    windowElement.add = function(el) {
```

Comment:
API-wise it's asymmetric to have `window.show()` modifying the window stack while `window.add()` modifies the `window`'s children.